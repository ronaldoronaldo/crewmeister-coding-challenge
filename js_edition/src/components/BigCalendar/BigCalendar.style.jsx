import styled from 'styled-components'
import {colors} from "../../styles/styleGuide";

export const CalendarWrapper = styled.div`
  padding-top: 20px;
  padding-bottom: 20px;
  background-color: ${colors.white};
`
