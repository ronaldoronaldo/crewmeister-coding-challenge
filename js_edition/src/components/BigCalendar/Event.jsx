import React from "react";
import { EventStyle } from "./Event.style.jsx";

export default function Event({ event }) {
  return (
    <EventStyle isSickness={event.type === "sickness"}>
      {` ${event.title} - ${event.type} `}
    </EventStyle>
  );
}
