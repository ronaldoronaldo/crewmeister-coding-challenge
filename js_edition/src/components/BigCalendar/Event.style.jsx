import styled from "styled-components";
import {colors, fontWeights, radius, spacing, text} from "../../config/styleGuide";

export const EventStyle = styled.div`
  ${({ isSickness }) => `
    border-radius: ${radius};
    background-color: ${isSickness ? colors.sickness : colors.vacation};
    padding: ${spacing.xSmall} ${spacing.small};
    font-weight: ${fontWeights.bold};
    color: ${colors.white};
    font-size: ${text.paragraph};
    margin-right: ${spacing.small};
  `};
`;
