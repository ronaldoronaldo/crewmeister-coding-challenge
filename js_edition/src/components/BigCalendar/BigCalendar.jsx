import React from "react";
import { Calendar, momentLocalizer, Views } from "react-big-calendar";
import moment from "moment";
import Event from "./Event/Event";
import {CalendarWrapper} from "./BigCalendar.style";
/* eslint-disable no-unused-vars */
import CalendarStyle from "react-big-calendar/lib/css/react-big-calendar.css";

const allViews = Object.keys(Views).map((k) => Views[k]);

export default function BigCalendar({ events }) {
  const localizer = momentLocalizer(moment);
  return (
    <CalendarWrapper>
      <Calendar
        localizer={localizer}
        events={events}
        views={allViews}
        startAccessor="start"
        endAccessor="end"
        defaultDate={new Date(2017, 0, 1)}
        style={{ height: 600 }}
        components={{
          eventWrapper: Event,
          toolbar: ''
        }}
      />
    </CalendarWrapper>
  );
}
