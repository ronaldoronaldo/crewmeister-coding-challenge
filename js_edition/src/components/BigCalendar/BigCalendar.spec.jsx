import React from "react";
import { render } from "@testing-library/react";
import moment from "moment";

import BigCalendar from "../BigCalendar";

describe("<BigCalendar />", () => {
  it("should match snapshot", () => {
    const events = [
      {
        id: 1,
        title: "Mike",
        allDay: true,
        type: "sickness",
        start: moment("2017-01-01"),
        end: moment("2017-01-01"),
      },
      {
        id: 2,
        title: "Tom",
        allDay: true,
        type: "vacation",
        start: moment("2017-01-02"),
        end: moment("2017-01-04"),
      },
    ];
    const { container } = render(<BigCalendar events={events} />);
    expect(container).toMatchSnapshot();
  });
});
