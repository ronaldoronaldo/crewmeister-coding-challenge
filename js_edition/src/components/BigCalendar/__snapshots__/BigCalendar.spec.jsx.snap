// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BigCalendar /> should match snapshot 1`] = `
<div>
  <div
    class="sc-AxhCb djqnBx"
  >
    <div
      class="rbc-calendar"
      style="height: 600px;"
    >
      <div
        class="sc-Axmtr kOaucV"
      >
        <div
          class="sc-AxheI kNCKog"
        >
          <button
            class="sc-AxgMl sc-AxmLO MIcGi"
          >
            Today
          </button>
          <button
            class="sc-AxgMl sc-AxmLO MIcGi"
          >
            Back
          </button>
          <button
            class="sc-AxgMl sc-AxmLO MIcGi"
          >
            Next
          </button>
        </div>
        <span
          class="sc-fzozJi fjaCUm"
        >
          <b>
            January
          </b>
           
          2017
        </span>
        <div
          class="sc-AxheI kNCKog"
        >
          <button
            class="sc-AxgMl sc-AxmLO MIcGi"
          >
            Day
          </button>
          <button
            class="sc-AxgMl sc-AxmLO MIcGi"
          >
            Week
          </button>
          <button
            class="sc-AxgMl sc-AxmLO MIcGi"
          >
            Month
          </button>
        </div>
      </div>
      <div
        class="rbc-month-view"
      >
        <div
          class="rbc-row rbc-month-header"
        >
          <div
            class="rbc-header"
          >
            <span>
              Sun
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Mon
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Tue
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Wed
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Thu
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Fri
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Sat
            </span>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell rbc-current"
              >
                <a
                  href="#"
                >
                  01
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  02
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  03
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  04
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  05
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  06
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  07
                </a>
              </div>
            </div>
            <div
              class="rbc-row"
            >
              <div
                class="rbc-row-segment"
                style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
              >
                <div
                  class="sc-AxjAm eAGnuO"
                >
                  <span
                    class="sc-AxirZ cOgLMt"
                  >
                    Mike
                  </span>
                  <span
                    class="sc-AxirZ sc-AxiKw iLSsjr"
                  >
                     - 
                    sickness
                  </span>
                </div>
              </div>
              <div
                class="rbc-row-segment"
                style="flex-basis: 28.57142857142857%; max-width: 28.57142857142857%;"
              >
                <div
                  class="sc-AxjAm iqqVjM"
                >
                  <span
                    class="sc-AxirZ cOgLMt"
                  >
                    Tom
                  </span>
                  <span
                    class="sc-AxirZ sc-AxiKw iLSsjr"
                  >
                     - 
                    vacation
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  08
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  09
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  10
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  11
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  12
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  13
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  14
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  15
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  16
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  17
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  18
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  19
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  20
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  21
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  22
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  23
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  24
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  25
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  26
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  27
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  28
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  29
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  30
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  31
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
              >
                <a
                  href="#"
                >
                  01
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
              >
                <a
                  href="#"
                >
                  02
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
              >
                <a
                  href="#"
                >
                  03
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
              >
                <a
                  href="#"
                >
                  04
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
